### PR TITLE
DDF-4594 Added ability to perform an action on multiple selected items

### PR DESCRIPTION
**Pull request by hayleynorton**
#### What does this PR do?
Creates a section above the visualization tabs for buttons that perform actions on multiple selected items. Currently the only button is "Export"
#### Who is reviewing it? 
@andrewzimmer @Bdthomson 
#### Ask 2 committers to review/merge the PR and tag them here.
@andrewkfiedler
@bdeining 
@djblue 
#### How should this be tested?
- Verify that an `Export` button exists over the visualization tabs and is disabled with no results selected
- Select at least one result and verify that the `Export` button is enabled
- Click the `Export` button and verify that the selected result(s) are properly exported
#### What are the relevant tickets?
Fixes: #4594 
#### Screenshots
DDF-4594-min (1)
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests
#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/multi-select-actions/multi-select-actions.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+`
+
+const Root = styled.div`
+  border-left: 1px solid
+    ${props =>
+      transparentize(0.9, readableColor(props.theme.backgroundContent))};
+`
+
+const Export = (props: Props) => (
+  <MultiSelectButton
```
Comment (samuelechu) [inline ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/multi-select-actions/multi-select-actions.tsx]: You can have two different styled buttons (enabled/disabled) to make things simpler. Example: https://github.com/codice/ddf/blob/d53daba2173bc9396948e2c949308564aeca2cf1/ui/packages/catalog-ui-search/src/main/webapp/react-component/layer-item/presentation/alpha.tsx

File: ui/packages/catalog-ui-search/src/main/webapp/plugins/multi-select-actions.tsx [hunk 2]
```diff
@@ -10,10 +9,5 @@
...
  **/
- --}}
-<div class="content-left">
 
-</div>
-<div class="content-right">
-
-</div>
\ No newline at end of file
+export default (multiSelectButtons: any[]) => multiSelectButtons
```
Comment (samuelechu) [inline ui/packages/catalog-ui-search/src/main/webapp/plugins/multi-select-actions.tsx]: why is this necessary? Is it a type issue?

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/container/multi-select-actions/multi-select-actions.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+class MultiSelectActions extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props)
+    this.state = {
+      isDisabled: true,
+    }
+  }
+
+  isDisabled = () => {
```
Comment (samuelechu) [inline ui/packages/catalog-ui-search/src/main/webapp/react-component/container/multi-select-actions/multi-select-actions.tsx]: I think it would be good to rename this function to avoid confusion between state and this function

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/multi-select-actions/multi-select-actions.tsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  </MultiSelectButton>
+)
+
+const buttons = plugin([Export])
+
+const render = (props: Props) => {
+  return (
+    <Root>
+      {buttons.map((Component: any, i: number) => (
+        <Component key={i} {...props} />
```
Comment (samuelechu) [inline ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/multi-select-actions/multi-select-actions.tsx]: Is it alright to use the index as the key in this situation? https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318

Comment (Bdthomson) [reply ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/multi-select-actions/multi-select-actions.tsx]: We do this in a couple places for extensibility, it's generally fine as long as the list isn't expected to change. However, if you wanted to be extra safe, you could do this - https://github.com/codice/ddf/blob/master/ui/packages/catalog-ui-search/src/main/webapp/react-component/container/metacard-interactions/index.tsx#L106-L110

File: ui/packages/catalog-ui-search/src/main/webapp/plugins/multi-select-actions.tsx (hunk 2 above)
Comment (Bdthomson) [reply ui/packages/catalog-ui-search/src/main/webapp/plugins/multi-select-actions.tsx]: What what you write here instead?

Comment (samuelechu) [reply ui/packages/catalog-ui-search/src/main/webapp/plugins/multi-select-actions.tsx]: I'm not really sure what this is used for, can it be deleted?

Comment (hayleynorton) [reply ui/packages/catalog-ui-search/src/main/webapp/plugins/multi-select-actions.tsx]: The purpose is to make the multi-select button region extensible

File: ui/packages/catalog-ui-search/src/main/webapp/component/content/content.view.js [hunk 5]
```diff
@@ -33,8 +33,62 @@ const QueryTitleView = require('../query-title/query-title.view.js')
 const GoldenLayoutView = require('../golden-layout/golden-layout.view.js')
 const LoadingCompanionView = require('../loading-companion/loading-companion.view.js')
 
+import { ChangeBackground } from '../../react-component/styles/mixins/change-background'
+import MultiSelectActions from '../../react-component/container/multi-select-actions'
+import styled from '../../react-component/styles/styled-components'
+
+const ContentLeft = styled.div`
+  ${props => {
```
Comment (bakejeyner) [inline ui/packages/catalog-ui-search/src/main/webapp/component/content/content.view.js]: ❓ Just curious: does this return css elements given the passed in `props`?

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/container/multi-select-actions/index.tsx [hunk 6]
```diff
@@ -0,0 +1,12 @@
+/**
```
Comment (bakejeyner) [inline ui/packages/catalog-ui-search/src/main/webapp/react-component/container/multi-select-actions/index.tsx]: ❓ Is there a reason this class is the same as https://github.com/codice/ddf/pull/4595/files#diff-f45927c091638e6fd42db33875f8bd08R12?

Is it because as we add functionality in the future, these two classes will start to look different?

File: ui/packages/catalog-ui-search/src/main/webapp/component/content/content.view.js (hunk 5 above)
Comment (samuelechu) [reply ui/packages/catalog-ui-search/src/main/webapp/component/content/content.view.js]: yep, it's a mixin: https://github.com/codice/ddf/blob/master/ui/packages/catalog-ui-search/src/main/webapp/react-component/styles/mixins/change-background.js

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/container/multi-select-actions/index.tsx (hunk 6 above)
Comment (samuelechu) [reply ui/packages/catalog-ui-search/src/main/webapp/react-component/container/multi-select-actions/index.tsx]: In general, we're having each React component have a container part, which deals with business logic, and a presentational part, which only cares about how the data looks. The container component will pass props to the presentational component. https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0

I was thinking that it might be good to keep the same concept, but instead group all the files related to a component in the same place.

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/container/multi-select-actions/multi-select-actions.tsx (hunk 3 above)
Comment (hayleynorton) [reply ui/packages/catalog-ui-search/src/main/webapp/react-component/container/multi-select-actions/multi-select-actions.tsx]: updateDisabled

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/multi-select-actions/multi-select-actions.tsx [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+const disabledStyle = {
+  cursor: 'not-allowed',
+  opacity: 0.3,
+}
+
+const Export = (props: Props) => (
+  <MultiSelectButton
+    style={props.isDisabled ? disabledStyle : {}}
+    buttonType={buttonTypeEnum.neutral}
+    onClick={props.handleExport}
```
Comment (hayleynorton) [inline ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/multi-select-actions/multi-select-actions.tsx]: disable click

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/multi-select-actions/multi-select-actions.tsx [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+  min-width: calc(2.5 * ${props => props.theme.minimumButtonSize});
+  padding: 0px 5px 0px 5px;
+`
+
+const disabledStyle = {
+  cursor: 'not-allowed',
+  opacity: 0.3,
+}
+
+type MultiSelectActionProps = {
```
Comment (willwill96) [inline ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/multi-select-actions/multi-select-actions.tsx]: ✏️ It's a little confusing to have `Props` and `MultiSelectActionProps` in the same file

**Comment by hayleynorton** [comment]
build now

**Comment by nsuvarna** [comment]
Hero in progress

**Comment by Bdthomson** [comment]
build now

**Comment by Bdthomson** [comment]
build now

**Comment by Bdthomson** [comment]
build now
